asyncpg/protocol/protocol.pyx

```diff
@@ -592,12 +704,7 @@ cdef class BaseProtocol(CoreProtocol):
         if self.cancel_waiter is not None:
             # We have received the result of a cancelled operation.
             # Check that the result waiter (if it exists) has a result.
-            if self.waiter is not None and not self.waiter.done():
-                self.cancel_waiter.set_exception(
-                    RuntimeError(
-                        'invalid result waiter state on cancellation'))
-            else:
-                self.cancel_waiter.set_result(None)
+            self.cancel_waiter.set_result(None)
```

Comment:
Did you intentionally remove that code? If so the comment above it needs to be adjusted.